lib/auth-header.js

```diff
@@ -7,10 +7,10 @@ module.exports = ({ scheme, token, username, password }) => {
     return `Token ${replace(token, config)}`;
   }
   if (scheme === 'basic') {
-    const basicAuth = [
-      replace(username, config),
-      replace(password, config),
-    ].join(':');
-    return `Basic ${Buffer.from(basicAuth).toString('base64')}`;
+    return `Basic ${Buffer.from(
+      token
+        ? replace(token, config) // Not standardized Basic auth format, currently using in Azure Repos API
+        : `${replace(username, config)}:${replace(password, config)}`,
```

Comment:
I think this would be more readable if we extracted things so the comment and ternary don't muddy the logic as much e.g.
```
const basicAuth = token ? getTokenBasedBasicAuth(token, config) : `${replace(username, config)}:${replace(password, config)}`;
return `Basic ${basicAuth}`;
```

Option to also move the basic auth to a separate function, though it didn't seem to add much.

Reply:
Not sure why we need `getTokenBasedBasicAuth` since it is one line: `replace(token, config)`

Reply:
It doesn't make a huge difference, but it's quicker to process a readable function name than some logic and a comment i.e. a ternary with `getTokenBasedBasicAuth() : getBasicAuth()` is quicker to grok than interpreting the logic that they constitute, let alone the comment which contains info that isn't always immediately necessary.